src/DeviceNameManager.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Scott Peshak
+ * Copyright 2013 BrewPi/Elco Jacobs.
+ *
+ * This file is part of BrewPi.
+ *
+ * BrewPi is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BrewPi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BrewPi.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <FS.h>
+#if defined(ESP32)
+#include <SPIFFS.h>
+#endif
+
+#include "DeviceNameManager.h"
+
+/**
+ * Set a human readable name for a device.
+ * @param device - The identifier for the device, most commonly the OneWire device address (in hex)
+ * @param name - The name to set
+ */
+void DeviceNameManager::setDeviceName(const char* device, const char* name)
+{
+  char filename[32];
+  DeviceNameManager::deviceNameFilename(filename, device);
+
+  File f = SPIFFS.open(filename, "w");
```

Comment:
At some point (later!) we should probably revisit how the device definitions are written out and see if they can also be refactored into JSON writes. Project for later!